src/engine/Account-Imap.cpp

```diff
@@ -165,17 +165,386 @@ std::cout << "delete_account " << category << ", " << key << std::endl;
+
+    /* fill in the account guid and number of tokens found for this account guid */
+    this_account->account_guid = (char*)key;
+    this_account->token_count = g_value_get_int64(value);
+
+    /* append onto the glist a pointer to the new account_token_count structure */
+    tokenInfo->accounts = g_list_prepend(tokenInfo->accounts, this_account);
+}
+
+/** intermediate values used to calculate the bayes probability of a given account
+  where p(AB) = (a*b)/[a*b + (1-a)(1-b)], product is (a*b),
+  product_difference is (1-a) * (1-b)
+ */
+struct account_probability
+{
+    double product; /* product of probabilities */
+    double product_difference; /* product of (1-probabilities) */
+};
+
+/** convert a hash table of account names and (struct account_probability*)
+  into a hash table of 100000x the percentage match value, ie. 10% would be
+  0.10 * 100000 = 10000
+ */
+#define PROBABILITY_FACTOR 100000
+static void
+buildProbabilities(gpointer key, gpointer value, gpointer data)
+{
+    GHashTable *final_probabilities = (GHashTable*)data;
+    struct account_probability *account_p = (struct account_probability*)value;
+
+    /* P(AB) = A*B / [A*B + (1-A)*(1-B)]
+     * NOTE: so we only keep track of a running product(A*B*C...)
+     * and product difference ((1-A)(1-B)...)
+     */
+    gint32 probability =
+        (account_p->product /
+         (account_p->product + account_p->product_difference))
+        * PROBABILITY_FACTOR;
+
+    PINFO("P('%s') = '%d'", (char*)key, probability);
+
+    g_hash_table_insert(final_probabilities, key, GINT_TO_POINTER(probability));
+}
+
+/** Frees an array of the same time that buildProperties built */
+static void
+freeProbabilities(gpointer key, gpointer value, gpointer data)
+{
+    /* free up the struct account_probability that was allocated
+     * in gnc_account_find_account_bayes()
+     */
+    g_free(value);
+}
+
+/** holds an account guid and its corresponding integer probability
+  the integer probability is some factor of 10
+ */
+struct account_info
+{
+    char* account_guid;
+    gint32 probability;
+};
+
+/** Find the highest probability and the corresponding account guid
+    store in data, a (struct account_info*)
+    NOTE: this is a g_hash_table_foreach() function for a hash table of entries
+    key is a  pointer to the account guid, value is a gint32, 100000x
+    the probability for this account
+*/
+static void
+highestProbability(gpointer key, gpointer value, gpointer data)
+{
+    struct account_info *account_i = (struct account_info*)data;
+
+    /* if the current probability is greater than the stored, store the current */
+    if (GPOINTER_TO_INT(value) > account_i->probability)
+    {
+        /* Save the new highest probability and the assoaciated account guid */
+        account_i->probability = GPOINTER_TO_INT(value);
+        account_i->account_guid = (char*)key;
+    }
+}
+
+
+#define threshold (.90 * PROBABILITY_FACTOR) /* 90% */
```

Comment:
Use ``` 
static const double CONFIDENCE {0.90};
constexpr double threshold {CONFIDENCE * PROBABILITY_FACTOR};
```
That way the compiler does the calculation once instead of a CPP macro that causes it to be done at run-time every time ```threshold``` is used.